### ISSUE TITLE
multi_scan will crash on re-runs when test identifiers has spaces in the name and older version of fastlane

### ISSUE THREAD

**Issue by lyndsey-ferguson** [lyndsey-ferguson/fastlane-plugin-test_center#130]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

### Issue Description

When a test identifier has a space in its name and the installed version of the fastlane is older than `2.114.0`, retries of a test will crash with:

```
xcodebuild: error: Unknown build action '<SecondWordInTestIdentifier>'
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  Here is the crash:
  <pre>
/Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane_core/lib/fastlane_core/ui/interface.rb:163:in `build_failure!': [0;31;49m[!] Error building/testing the application - see the log above[0m (FastlaneCore::Interface::FastlaneBuildFailure)
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/scan/lib/scan/error_handler.rb:34:in `handle_build_error'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/scan/lib/scan/runner.rb:50:in `block in test_app'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane_core/lib/fastlane_core/command_executor.rb:88:in `execute'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/scan/lib/scan/runner.rb:42:in `test_app'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/scan/lib/scan/runner.rb:22:in `run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:54:in `run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:40:in `run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/helper/multi_scan_manager/test_batch_worker.rb:18:in `run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:143:in `block in run_test_batches'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `each'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `each_with_index'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `run_test_batches'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:53:in `run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bundler/gems/fastlane-plugin-test_center-4e045d37def3/lib/fastlane/plugin/test_center/actions/multi_scan.rb:23:in `run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile.rb:37:in `block (2 levels) in parsing_binding'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/fastlane-2.106.2/bin/fastlane:23:in `<top (required)>'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bin/fastlane:22:in `load'
	from /Users/jenkins/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bin/fastlane:22:in `<main>'

  </pre>

  Here is what is being passed to `scan`:

<pre>
only_testing               | ["Lyndsey SecondWordInTestIdentifier Tests/PuppetTests/testThatThumbArticulates"] 

</pre>
</details>

### Environment

<details>
  <pre>[INSERT OUTPUT HERE]</pre>
</details>

**Comment by jineshqa** [comment]
Hi @lyndsey-ferguson 

Are you referring to this error?

`/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rexml/parent.rb:58:in `[]': [!] no implicit conversion of String into Integer (TypeError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/rexml/parent.rb:58:in `[]'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/actions/collate_junit_reports.rb:114:in `increment_testable_tries'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/actions/collate_junit_reports.rb:17:in `block in run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/actions/collate_junit_reports.rb:16:in `each'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/actions/collate_junit_reports.rb:16:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/report_collator.rb:58:in `collate_junit_reports'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/report_collator.rb:21:in `collate'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan_helper.rb:139:in `collate_reports'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan_helper.rb:120:in `after_testrun'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:58:in `rescue in run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:46:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/retrying_scan.rb:40:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/test_batch_worker.rb:18:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:143:in `block in run_test_batches'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `each'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `each_with_index'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:140:in `run_test_batches'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:53:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-plugin-test_center-3.8.0/lib/fastlane/plugin/test_center/actions/multi_scan.rb:23:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:595:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.107.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:22:in `load'
	from /usr/local/bin/fastlane:22:in `<main>'
Build step 'Execute shell' marked build as failure`

**Comment by lyndsey-ferguson** [comment]
Hi @jineshqa, no, the issue that I reported occurs within `multi_scan` when it sends the `only_testing` parameter to `scan` which then sends it to `xcodebuild`. I'll update the description.

The crash log that you posted indicates that there is a problem when collating the junit reports, particularly, when the retries attribute is being updated. Could you create a new issue please? It would help if you could include two junit report files that, when passed to the action `collate_junit_reports` causes this problem.

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.8.1](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.8.1)
